src/pegen/parser.py

```diff
@@ -263,16 +272,19 @@ def positive_lookahead(self, func: Callable[..., T], *args: object) -> T:
         self._reset(mark)
         return ok
 
-    def negative_lookahead(self, func: Callable[..., object], *args: object) -> bool:
+    def negative_lookahead(self, func: Callable[..., object], *args: object) -> Union[Literal[True], NotParsedMarker]:
         mark = self._mark()
         ok = func(*args)
         self._reset(mark)
-        return not ok
+        return not_parsed if ok is not not_parsed else True
```

Comment:
bool check is more than enough here and in positive_lookahead

Reply:
No, `False` now a correct parsing result. For mark incorrect parsing result we always should use `not_parsed`.